File: src/gestorAplicacion/clasesHerencia/Ejecutivo.java
package gestorAplicacion.clasesHerencia;
import java.util.ArrayList;


import gestorAplicacion.clasesPrincipales.*;

public class Ejecutivo extends Bus {
	private final int precioKm=500;//precio por km
	int codigo;
	private final int capacidad=26;
	public static ArrayList<Ejecutivo> ejecutivos = new ArrayList<Ejecutivo>();
	
	//constructor
	public Ejecutivo(String placa, Conductor conductor,ArrayList<String> sillas) {
		super(placa,conductor,sillas);
		this.codigo=150;
		Ejecutivo.ejecutivos.add(this);
	}
	
	public Ejecutivo() {
		
	}
	
	//get_precio
	public int getPrecioKm() {
		return precioKm;
	}
	//get&set codigo
	public int getCodigo() { //desde aqui agregue 
		return codigo;
	}

	public void setCodigo(int codigo) {
		this.codigo = codigo;
	}
	@Override
	public void sillasDisponibles() {
		for(Tiquete tiquete : Viaje.getTiquetesTodos()){
			if(tiquete.getEstado() == false){
				System.out.println("la silla: "+tiquete.getSillaTiquete()+"esta disponible");
				}
			}
		}
	@Override
	public String sillaTiquete(Tiquete tiquete) {
		return tiquete.getSillaTiquete();
	}
}

File: Python/Ejecutivo.py
from Bus import Bus
from Viaje import Viaje

class Ejecutivo(Bus):

    def __init__(self,placa,conductor,sillas,precioKm=500,codigo=150,capacidad=26):
        ejecutivos = []
        super().__init__(placa,conductor,sillas)
        Ejecutivo.ejecutivos.append(self)

    #get&set capacidad
    def getCapacidad(self):
        return self._capacidad
    def setCapacidad(self, capacidad):
        self._capacidad = capacidad

    #get_precio
    def getPrecioKm(self):
        return self._precioKm
    def setPrecio(self, precioKm):
        self._precioKm = precioKm

    #get&set codigo
    def getCodigo(self):
        return self.codigo
    def setCodigo(self, codigo):
        self.codigo = codigo

    #Metodo_sillasdisponibles
    def sillasDisponibles(self):
        for tiquete in Viaje.getTiquetesTodos():
            if tiquete.getEstado() == False:
                print("la silla: "+tiquete.getSillaTiquete()+"esta disponible")
    def sillaTiquete(self, tiquete):
        return tiquete.getSillaTiquete()
